webextension/background.js

```diff
@@ -467,13 +481,29 @@ async function handleButtonClick(action, tabState) {
   }
 }
 
+async function updatePageActionIcon(tabId) {
+  const active = (gCurrentlyPromptingTab || {}).id === tabId;
+  const path = active ? "icons/broken_page_active.svg"
+                      : "icons/broken_page.svg";
+  await browser.pageAction.setIcon({tabId, path});
+}
+
 function getActiveTab() {
   return browser.tabs.query({active: true, lastFocusedWindow: true}).then(tabs => {
     return tabs[0];
   });
 }
 
-function closePopup() {
+async function showPageAction(tabId) {
+  /* return new Promise(resolve => {
+     requestAnimationFrame(async function() {
```

Comment:
`rAF` is interesting here. What's it doing for us? (or, will it do, once this part is implemented).